### PR TITLE
feat: add mobilize dbt tests

### PR DESCRIPTION
**Pull request by jitoquinto**
@emily-flambe what do we do if we run tests and dev and find dupes 🙃 ?? Should i run in prod and see if its a problem there? 

Anyway, here are the dbt tests for mobilize! this is my first time doing this so let me know if I'm missing anything or if i did something wrong!

### CODE REVIEW

File: dbt-cta/mobilize/models/1_cta_full_refresh/_1_cta_full_refresh__models.yml [hunk 1]
```diff
@@ -0,0 +1,804 @@
+version: 2
+
+models:
+
+  - name: affiliations_base
+    description: ''
+    columns:
+      - name: id
```
Comment (kanelouise) [inline dbt-cta/mobilize/models/1_cta_full_refresh/_1_cta_full_refresh__models.yml:8]: I think it could be useful to add uniqueness tests on the primary keys of these non-ab base tables 🤷

**Review by kanelouise — APPROVED** [review]
I found nothing to flag. Looks good, my one suggestion would be to add primary key tests to the base tables that are not an AB ID. Redundant? Maybe. But it could also reveal a dupe where the AB ID is different but the table primary key is duplicated.

File: dbt-cta/mobilize/models/1_cta_full_refresh/_1_cta_full_refresh__models.yml (hunk 1 above)
Comment (emily-flambe) [reply dbt-cta/mobilize/models/1_cta_full_refresh/_1_cta_full_refresh__models.yml:8]: agreed! logically the `*_hashid` fields _should_ also function as primary keys, but it seems like a good idea to also test for uniqueness when there's another field that seems like it ought to be the obvious PK for the table (I think generally a safe assumption when it's a field called `id`). Even if those tests should be redundant, it can still be useful to have them both - and if that reveals that an "obvious" UPK is not actually unique, that might be something we document somewhere (maybe the README for the dbt project).

Bottom line, I agree with @kanelouise, let's add non-null and unique tests to all the `id` fields 😎

File: utils/cta_dbt_helper.sh [hunk 2]
```diff
@@ -131,7 +131,7 @@ generate_dbt_tests() {
         INPUT_DIR_NAME=$(gum input --prompt "Enter the name of the vendor to generate tests for: " --placeholder "(ex. actblue)")
         OPTION_UNIVERSAL_TESTS=$(gum input --prompt "Would you like to initialize these files using the default tests (see universal_tests.yml)? " --placeholder " (Y is recommended! Or leave blank to skip)")
         if [ "$OPTION_UNIVERSAL_TESTS" == "Y" ]; then
-            CLI_OPTIONS="--universal-tests-path ../utils/universal_tests.yml"
```
Comment (emily-flambe) [inline utils/cta_dbt_helper.sh:134]: Ooh thank you for making this change, definitely more legit!

**Review by emily-flambe — DISMISSED** [review]
I agree with kane's suggestion of adding uniqueness and not-null tests to the `id` columns for each table, though that's not technically a blocker! Congrats on your first dbt test experience, bonus points awarded for including source tests as well (I sometimes forget about those 🫠

**Comment by jitoquinto** [comment]
> I found nothing to flag. Looks good, my one suggestion would be to add primary key tests to the base tables that are not an AB ID. Redundant? Maybe. But it could also reveal a dupe where the AB ID is different but the table primary key is duplicated.

Thats a good point! I agree that if we can find a primary key that isnt AB generated, we should! def have ran into the case where AB ID was unique but the row itself wasnt 🙃. I'll look through and add PK tests where they seem appropriate!

File: utils/cta_dbt_helper.sh (hunk 2 above)
Comment (jitoquinto) [reply utils/cta_dbt_helper.sh:134]: yeah im hoping to make this script not care what directory you run it from!

File: dbt-cta/mobilize/models/1_cta_full_refresh/_1_cta_full_refresh__models.yml (hunk 1 above)
Comment (jitoquinto) [reply dbt-cta/mobilize/models/1_cta_full_refresh/_1_cta_full_refresh__models.yml:8]: makes sense! ill give it a go and try to find PKs to add tests to :catsalute:

**Comment by jitoquinto** [comment]
@emily-flambe @kanelouise added those uniqueness and not null tests to `id` fields! I ran the tests for a few of the sv_mobilize_org datasets in prod and they all passed 🎉  so those id's are indeed PKs

**Comment by emily-flambe** [comment]
> @emily-flambe @kanelouise added those uniqueness and not null tests to `id` fields! I ran the tests for a few of the sv_mobilize_org datasets in prod and they all passed 🎉 so those id's are indeed PKs

hell yea 🙌 ship it!

**Comment by jitoquinto** [comment]
> > @emily-flambe @kanelouise added those uniqueness and not null tests to `id` fields! I ran the tests for a few of the sv_mobilize_org datasets in prod and they all passed 🎉 so those id's are indeed PKs
> 
> hell yea 🙌 ship it!

:catjam: @emily-flambe could you re-review?

**Review by emily-flambe — APPROVED** [review]
ope!
